### PR TITLE
test: Make the unlikely race in p2p_invalid_messages impossible

### PR DESCRIPTION
**Pull request by maflcko**
After `add_p2p_connection` both sides have the verack processed.
However the pong from conn in reply to the ping from the node has not
been processed and recorded in totalbytesrecv.
Flush the pong from conn by sending a ping from conn.

This should make the unlikely race impossible.

### CODE REVIEW

File: test/functional/p2p_invalid_messages.py [hunk 1]
```diff
@@ -80,6 +80,11 @@ def run_test(self):
     def test_buffer(self):
         self.log.info("Test message with header split across two buffers is received")
         conn = self.nodes[0].add_p2p_connection(P2PDataStore())
+        # After add_p2p_connection both sides have the verack processed.
+        # However the pong from conn in reply to the ping from the node has not
+        # been processed and recorded in totalbytesrecv.
+        # Flush the pong from conn by sending a ping from conn.
```
Comment (pinheadmz) [inline test/functional/p2p_invalid_messages.py:86]: Is the flush on the node side? Like the pong has been sent but is still sitting in nodes buffer? I'm just trying to reliably break the test on master to examine the patch

**Review by pinheadmz — COMMENTED** [review]
concept ACK

Would it make any more sense to instead add `wait_for_verack=True` to the initial `add_p2p_connection()`? Looks like that would also call`sync_with_ping()` albeit with a longer timeout (default 60s).

**Comment by pinheadmz** [comment]
> Yes, this is the default value already.

Oh right -- why then does a second `sync_with_ping()` fix the race? Should `if wait_for_verack` wait for the node's ping before sending its own back?

**Review by mzumsande — COMMENTED** [review]
ACK fa4732afd23f67cd1bb45a60a66a7c628b64d560

Is there a reason for the question marks? 
To me, this looks like a straightforward fix: the explanation makes total sense (the "received: pong (...)" message occurs late in the log of the failed run), and so does the fix - the python conn sends another ping and waits for an answer, so the node must have also processed the earlier pong from its recv buffer for that happen.

**Comment by mzumsande** [comment]
ACK fa1eb0ecaef14d428812f956082d29ab134fc728

**Review by pinheadmz — APPROVED** [review]
ACK fa1eb0ecaef14d428812f956082d29ab134fc728

Thanks @mzumsande for the explanation, I compared the logs of failed and successful tests and I can see the message order is fixed

<details><summary>Show Signature</summary>

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

ACK fa1eb0ecaef14d428812f956082d29ab134fc728
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEE5hdzzW4BBA4vG9eM5+KYS2KJyToFAmQUfpsACgkQ5+KYS2KJ
yTrJUg//TlRg8JalWfEd+Rb2hklRFFJpE7PcaEajr5in71WlXaCQ9mCRaXHNtkI5
XepC5Ho7CCNt84FwRPZSr3giAh1W2S/ooJzKVJN2SOea2NdA1YzXJ1lFE1R84ylX
oADbsSHOVj49BNRORY6PwxTH5WgE2xCCe63h/I9cbazhBMPT7fKlJRYhP6jyEd7c
Z8dfhB68KB/XM8BEAqOlhsUedS4J41j/Xdaa3f9yYDNPuoHl4zjtPzjO+MMC1Zn8
/nhWQxVPe5B9QJEWOQLV7RXuE7StLbFm+2geJdZJv/1NYncx8Zl/VeiO1laKM2J3
nDBPA5pxhRMEi+u906fjU4ideGDC49SoR/k6SfPnbi9ZBx+cOvaBIZTDLQGXyCyU
UnlARR4XbogJ8Lmn5CAxAi6Csv1zpChYtnSjMhSqe5AEbmbHBH1/NJY/rkweOhbJ
Gf6+oo0oZme1yhrdZ8STa+kGNZaYVS239fmgOAoOdDWjnDZ/YGX1ms5EzzgqSW1F
PfNtaIDBWiXrJqHhNBf8NefG/GK4tkU5HlLstkE3RpvWlKBvHs6v/hheQnL0oKgo
wwsNU9Va9mZ3oxjIvJ/e8VST0a1WW3x5QZAnrNQYNrccySrZGfA35G80rYspaDuY
OWPFNSM9YtyhGlZii2MlPUG7DUVnh3nQ4bb868N9YQniPajLTA0=
=WS49
-----END PGP SIGNATURE-----
```

pinheadmz's public key is [on keybase](https://keybase.io/pinheadmz/pgp_keys.asc?fingerprint=e61773cd6e01040e2f1bd78ce7e2984b6289c93a)
</details>

**Pull request merged**